src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1034,6 +1034,7 @@ free_request(struct sol_http_request *request)
         switch (param->type) {
         case SOL_HTTP_PARAM_POST_DATA:
             free((char *)param->value.data.filename.data);
+            break;
```

Comment:
Don't need this break. It'll cause memory leak.
If you really want to change it you will need to do:

```
    SOL_HTTP_PARAMS_FOREACH_IDX (&request->params, param, idx) {                                               
        switch (param->type) {                                                                                 
        case SOL_HTTP_PARAM_POST_DATA:
            free((char*)param->value.data.value.data);                                                   
            free((char *)param->value.data.key.data);          
            free((char *)param->value.data.filename.data);                                                     
            break;                                                                                             
        case SOL_HTTP_PARAM_POST_FIELD:                                                                        
            free((char *)param->value.key_value.value.data);                                                   
            free((char *)param->value.key_value.key.data);                                                     
            break;                                                                                             
        default:                                                                                               
            break;                                                                                             
        }                                                                                                      
    }    
```

To my mind, just let it as it is.